core/src/main/scala/chisel3/Module.scala

```diff
@@ -238,14 +243,16 @@ package internal {
       val cloneParent = Module(new ModuleClone(proto))
       require(proto.isClosed, "Can't clone a module before module close")
       require(cloneParent.getOptionRef.isEmpty, "Can't have ref set already!")
+      // Chisel ports can be Data or Property, but to clone as a Record, we can only return Data.
+      val dataPorts = proto.getChiselPorts.collect { case (name, data: Data) => (name, data) }
```

Comment:
ClonePorts wants to refer to the ports of a Module as a Record, so I'm filtering out non-Data ports here, which seems like the only option if things like Property aren't Data.